lib/src/image_provider/_image_provider_io.dart

```diff
@@ -109,7 +114,11 @@ class CachedNetworkImageProvider
   @override
   bool operator ==(dynamic other) {
     if (other is CachedNetworkImageProvider) {
-      return url == other.url && scale == other.scale;
+      if (cacheKey != null && other.cacheKey != null) {
+        return cacheKey == other.cacheKey && scale == other.scale;
+      } else {
+        return url == other.url && scale == other.scale;
+      }
```

Comment:
We should do the same check as in the cachemanager, so if the key is not set use the url as key. If one of the keys is not set they can be different even though the url is the same.
```suggestion
      var sameKey = (cacheKey ?? url) == (other.cacheKey ?? other.url);
      return sameKey && scale == other.scale;
```